### ISSUE TITLE
problem importing msgack-python

### ISSUE THREAD

**Issue by arunpersaud** [dask/distributed#1913]
When I try to run dask-mpi I get the following error:

```
arun@apersaud:~>mpirun n0-4 dask-mpi --scheduler-file /home/$USER/scheduler.json
mpirun: cannot start dask-mpi on n1: invalid node
Traceback (most recent call last):
  File "/usr/bin/dask-mpi", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3088, in <module>
    @_call_aside
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3072, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3101, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 574, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 892, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 778, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'msgpack-python' distribution was not found and is required by distributed
```
However, I have msgpack-python installed.

```
arun@apersaud:~>python3
Python 3.6.4 (default, Jan 03 2018, 13:52:55) [GCC] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import msgpack
>>> msgpack.version
(0, 5, 6) 
```
I'm running linux (opensuse tumbleweed) with
python3-distributed version     : 1.21.6
and
python3-dask version     : 0.17.2

Any idea what is going wrong here?

**Comment by arunpersaud** [comment]
Hi, looked around a bit more and it seems that  msgpack-python got renamed to msgpack? see https://pypi.org/project/msgpack/
Could it be the that the name needs to be updated in distributed somewhere?

**Comment by lesteve** [comment]
tl;dr: either
* pip3 install msgpack-python until this issue is solved
* `pip3 install --only-binary :all: distributed`. You may need to update pip to do that (`pip install --upgrade pip`).

I quickly looked at this one and it seems like you get a different dask-mpi script whether you are installing 1.21.6 from the binary wheel or from the source .tar.gz.

### From the binary wheel

```
pip uninstall -y distributed
pip install --only-binary :all: distributed
hash -r && which dask-mpi
cat $(which dask-mpi)
```

```py
#!/home/local/lesteve/miniconda3/envs/tmp/bin/python

# -*- coding: utf-8 -*-
import re
import sys

from distributed.cli.dask_mpi import go

if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
    sys.exit(go())
```

### From the source .tar.gz

```
pip uninstall -y distributed
pip install --no-binary :all: distributed
hash -r && which dask-mpi
cat $(which dask-mpi)
```

```py
#!/home/local/lesteve/miniconda3/envs/tmp/bin/python
# EASY-INSTALL-ENTRY-SCRIPT: 'distributed==1.21.6','console_scripts','dask-mpi'
__requires__ = 'distributed==1.21.6'
import re
import sys
from pkg_resources import load_entry_point

if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
    sys.exit(
        load_entry_point('distributed==1.21.6', 'console_scripts', 'dask-mpi')()
    )
```

I am not an expert on packaging but i seem to have gathered that "avoid easy_install at all costs" is a general guideline. Not sure whether/how `setup.py` can be tweaked to avoid this.

**Comment by jakirkham** [comment]
There are also `conda` packages for it if you use `conda`.

```
conda install msgpack-python
```

**Comment by arunpersaud** [comment]
The pip install worked for me (don't have conda), so I can run my code for now... is there anything I can submit to fix it on the linux rpm side? Those are generated with python setup.py install or will this be fixed in the next release? Guess I still don't quite understand where the problem is.

**Comment by lesteve** [comment]
I will try to summarize and hopefully make this clearer: 
* in principle `distributed` should only care about being able to do `import msgpack`. It should not care that you did `pip install msgpack` (the new package name on PyPI) or `pip install msgpack-python` (the old package name on PyPI)
* for some obscure reason related to our `setup.py` (and easy_install), it looks like in some cases `distributed` does care about it and insist that you do `pip install msgpack-python`
* for now you can use work-arounds (e.g. pip install msgpack-python) but in principle this should be fixed in our `setup.py`

**Comment by mrocklin** [comment]
@lesteve can you recommend concrete changes that we should make to address this?

**Comment by lesteve** [comment]
I am not a packaging expert I am afraid so I don't know how to fix that on the `setup.py` ... It feels like renaming `msgpack-python` to `msgpack` in `requirements.txt` might just be a reasonable work-around. Also `msgpack` is the new name of the package after all.

Here is I think what will happen:
* `msgpack` will be installed from pip  and the `msgpack` folder will be overwritten in site-packages
* pkg_resources.load_entry_point will not raise an error because `msgpack` has the right name as far as setuptools entry point is concerned (I don't really understand this point, truth be told)

If you are switching between two distributed versions across the change you probably require both msgpack and msgpack-python, but that will happen automatically when you reinstall distributed. Having both `msgpack` and `msgpack-python` listed by `pip list` although confusing is probably not a great deal.

For completeness here is how the problem reported by the OP can happen:
* you have a working dask installation **that you installed from source and not from a wheel**
* you see that `msgpack-python` has been renamed to `msgpack`, so you do:
```
pip uninstall -y msgpack-python  # if you don't uninstall msgpack-python everything works
pip install msgpack
dask-scheduler  # The 'msgpack-python' distribution was not found and is required by distributed
```

**Comment by lesteve** [comment]
I opened #1927 to do the msgpack-python renaming.
